rfcs/text/0013_background_sessions.md

```diff
@@ -0,0 +1,484 @@
+// Any follow up searches executed with this sessionId will be saved into this object as well.  
+const sessionId = searchService.session.get();
+const backgroundSession = await searchService.sendToBackground(sessionId);
+```
+
+# Motivation
+
+Kibana is great at providing fast results from large sets of "hot" data. However, there is an increasing number of use cases where users want to analyze large amount of "colder" data (such as year-over-year reports, historical or audit data, batch queries, etc.). 
+
+For these cases, users run into two limitations: 
+ 1. Kibana has a default timeout of 30s per search. This setting is controlled by the `elasticsearch.requestTimeout` setting. Originally intended to protect clusters from unintentional overload by a single query.
+ 2. Kibana cancels queries upon navigating away from an applicaiton, once again, as a mean of protecting clusters and reducing unnecessary load. 
+
+In 7.7, with the introduction of the `_async_search` API in Elasticsearch, we provided Kibana users a way to bypass the timeout, but users still need to remain on-screen for the entire duration of the search requests.
+
+The primary motivation of this RFC is to enable users to do the following without needing to keep Kibana open, or while moving onto other work inside Kibana:
+
+- Run long search requests (beyond 30 seconds)
+- View their status (complete/incomplete)
+- Cancel incomplete search requests
+- Retrieve completed search request results 
+
+# Detailed design
+
+Because a single view (such as a dashboard with multiple visualizations) can initiate multiple search requests, we need a way to associate the search requests together in a single entity. 
+
+We call this entity a `session`, and when a user decides that they want to continue running the search requests while moving onto other work, we will create a saved object corresponding with that specific `session`, persisting the *sessionId* along with a mapping of each *request's hash* to the *async ID* returned by Elasticsearch.
+
+## High Level Flow Charts
+
+### Client side search
+
+This diagram matches any case where `data.search` is called from the front end:
+
+![image](../images/background_sessions_client.png)
+
+### Server side search
+
+This case happens if the server is the one to invoke the `data.search` endpoint, for example with TSVB.
+
+![image](../images/background_sessions_server.png)
+
+## Data and Saved Objects
+
+### Background Session Status
+
+```ts
+export enum BackgroundSessionStatus {
+  Running,   // The session has at least one running search ID associated with it.
+  Done,      // All search IDs associated with this session have completed.
+  Error,     // At least one search ID associated with this session had an error. 
+  Expired,   // The session has expired. Associated search ID data was cleared from ES.
+}
+```
+
+### Saved Object Structure
+
+```ts
+interface BackgroundSessionAttributes extends SavedObjectAttributes {
```

Comment:
Will the background session saved object be space-aware? If so, do you expect that a background session will be scoped to a specific space, or could a session span multiple spaces? My gut feeling says it should be scoped to a single space, but I don't have all of the necessary context to say for sure.

Reply:
Yes, the saved object should be scoped to a single space.